include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -884,6 +888,178 @@ namespace VectorTools
   }
 
 
+  template <int dim, typename VectorType, int spacedim, int fe_degree>
+  void project_parallel (const Mapping<dim, spacedim>   &mapping,
+                         const DoFHandler<dim,spacedim> &dof,
+                         const ConstraintMatrix         &constraints,
+                         const Quadrature<dim>          &quadrature,
+                         const std::function< typename VectorType::value_type (const typename DoFHandler<dim, spacedim>::active_cell_iterator &, const unsigned int)> func,
+                         VectorType                     &vec_result)
+  {
+    const parallel::Triangulation<dim,spacedim> *parallel_tria =
+      dynamic_cast<const parallel::Triangulation<dim,spacedim>*>(&dof.get_tria());
+    Assert (parallel_tria !=0, ExcNotImplemented());
```

Comment:
Why is this actually necessary? Wouldn't this function work just fine if the triangulation was sequential?